### PR TITLE
fix: preserve csp-nonce on mergeHead, warn on dropped streamed resolve

### LINKED ISSUE #1050: mergeHead replaces the csp-nonce meta on a background full-body swap
## Problem

`mergeHead` (the full remove+add head merge used ONLY on the background in-place full-body swap path, a snapshot restore or its revalidation) treats the live `<meta name="csp-nonce">` as stale whenever the incoming response carries a different per-request nonce (which it always does on a CSP-enabled app), removes it, and appends the incoming one. That violates the contract documented on `getCspNonce()`: the browser enforces CSP against the nonce the ORIGINAL page load declared, so after the merge `getCspNonce()` returns a nonce the active policy does not allow, and every later `cloneScriptWithCorrectNonce` / `cloneElementWithCorrectNonce`-stamped script or modulepreload is blocked.

This is the same bug class PR #1049 fixed for the boundary/frame paths (`reconcileHeadMetas` excludes `META_KEY_CSP_NONCE`), found while reviewing that PR. `mergeHead` predates #1049 and was not touched by it, so this is a pre-existing gap on the background path, filed separately.

Reach is narrow: CSP opt-in (`webjs.csp`) + a popstate snapshot restore's background revalidation (the only `mergeHead` caller) + a later dynamically-inserted nonce-stamped element. But when it hits, scripts silently stop executing until a hard reload.

## Design / approach

Give `mergeHead` the same csp-nonce carve-out `reconcileHeadMetas` has: skip `meta[name="csp-nonce"]` in BOTH its removal loop and its add loop (the incoming per-request nonce is useless client-side anyway, since the enforced header is the original page load's). `META_KEY_CSP_NONCE` / `metaIdentity` already exist in the same file, so the check can reuse them or just match the name attribute directly.

## Implementation notes (for the implementing agent)

- Where to edit: `packages/core/src/router-client.js` `mergeHead` (~L3895). The removal loop (~L3915-3928) and the add loop (~L3930-3943) both need the skip. `META_KEY_CSP_NONCE` + `metaIdentity` are defined ~L3741-3770.
- Landmine: the existing test `mergeHead: applies meta csp-nonce to created scripts (replaces source nonce)` (`packages/core/test/routing/router-client.test.js` ~L942) uses the SAME nonce on both sides, so it passes today and does not cover this. Do not weaken it; add the differing-nonce case alongside it.
- The counterfactual: live head `<meta name="csp-nonce" content="ORIGINAL">`, incoming head with `content="INCOMING"`, run `_mergeHead`, assert exactly one csp-nonce meta remains with `ORIGINAL` content. Fails before the fix (the meta is replaced).
- Invariants: `.js` + JSDoc only in `packages/`; #936 stylesheet preservation in `mergeHead` must stay untouched.
- Tests: node-level in `router-client.test.js` (the `_merge` harness export already exists). No browser layer strictly needed (pure head-merge logic, same as the existing mergeHead tests), but confirm nothing in the CSP browser suites asserts the opposite.

## Acceptance criteria

- [ ] After `mergeHead` with a differing incoming nonce, the live csp-nonce meta keeps the ORIGINAL content and no duplicate exists
- [ ] Scripts created after the merge get the original page-load nonce via `getCspNonce()`
- [ ] The existing same-nonce test still passes; a differing-nonce counterfactual fails when the fix is reverted

### LINKED ISSUE #1051: Dev-warn when a streamed Suspense resolve is dropped (absent boundary)
## Problem

`applyStreamedResolve` (the client router's streamed-Suspense applier) silently drops a resolve when its boundary placeholder `#sN` is not in the DOM (`if (!boundary) return;`). The drop itself is the CORRECT designed posture: every legitimately-reachable absent case is benign (a superseded nav fenced by `isCurrent()`, a discarded revalidation, a degrading full page load, a frame-missing no-swap). But the silence has a real cost: when a BUG wrongly makes a valid boundary absent, the only symptom is a mysteriously stuck skeleton with zero signal. That is exactly how the #1048 view-transition race stayed hidden and took a full live-repro investigation to diagnose. A one-line dev console warning would have surfaced it on the first glance.

This is an observability follow-up from the PR #1049 review discussion, NOT a behavior change. Do not add a retry or apply-anyway path: a retry would run outside the streamer's nav-token fence and could splice a superseded nav's content into a recycled boundary id (boundary ids are per-render sequential).

## Design / approach

Mirror the router's existing degradation-diagnostics pattern, `devWarnFallback` (`packages/core/src/router-client.js` ~L775): dev-only (`process.env.NODE_ENV === 'production'` early-return), `warnOnce`-keyed so a burst of dropped boundaries does not spam, message naming the dropped boundary id and the likely causes, e.g. `[webjs] dropped streamed Suspense resolve "s1": no #s1 boundary in the DOM. Benign if this navigation was superseded or degraded; a stuck skeleton here means the shell swap did not place the boundary.`

Keep the drop itself byte-identical (return unchanged). The warning goes in the `!boundary` branch only.

## Implementation notes (for the implementing agent)

- Where to edit: `packages/core/src/router-client.js` `applyStreamedResolve` (~L4093, the `if (!boundary) return;` branch, which already carries a comment block explaining why the drop is deliberate; extend it rather than contradicting it). `devWarnFallback` (~L775) and `warnOnce` are in the same file and show the exact dev-gating + once-keying idiom; key the warn on the boundary id (`stream-drop:${id}`).
- Landmines:
  - Do NOT warn on the `isCurrent()`-fenced skips in `streamBoundariesProgressively` (~L4110): a superseded nav's skipped applies are routine, not diagnostic. Only the reached-the-DOM-and-found-no-boundary case warns.
  - `_resetWarnOnce` is exported for tests (`packages/core/test/routing/router-client.test.js` uses it); reuse it so the new warn is testable deterministically.
  - The function also runs during background revalidation paths; the warn must never throw and never run in production (the `devWarnFallback` gate handles both).
- Invariants: `.js` + JSDoc in `packages/`; no behavior change to the drop; client-router change, so a browser assertion is preferred, but a pure console-warn addition is observable in the node/linkedom harness too (stub `console.warn`), which matches how other warn-once tests in `router-client.test.js` assert.
- Tests: node test asserting (a) a dropped resolve emits exactly one dev warning naming the id, (b) a second drop of the same id does not re-warn, (c) `NODE_ENV=production` emits nothing, and (d) a successful resolve emits nothing. Counterfactual: the test fails when the warn line is removed.
- Docs: none (dev-only diagnostics; no public surface change). No scaffold/MCP/editor surface.

## Acceptance criteria

- [ ] A dropped streamed resolve logs one dev-only warning naming the boundary id and pointing at the stuck-skeleton implication
- [ ] No warning in production, on superseded-nav fenced skips, or on successful resolves; no repeat spam for the same id
- [ ] The drop behavior itself is byte-identical (no retry, no apply-anyway)
- [ ] A counterfactual proves the test fires (removing the warn reds it)

### PR DESCRIPTION
**Pull request by vivek7405**
Two small client-router fixes found during the #1049 review, both in `packages/core/src/router-client.js`.

## Closes #1050 - mergeHead replaced the csp-nonce meta on a background full-body swap

`mergeHead` (used only on the background in-place full-body path: a popstate snapshot restore or its revalidation) removed the live `<meta name="csp-nonce">` and appended the incoming one whenever the per-request nonce differed (which it always does on a CSP-enabled app). That violates `getCspNonce()`'s contract, the browser enforces CSP against the nonce the ORIGINAL page load declared, so after the merge later `cloneScriptWithCorrectNonce` / `cloneElementWithCorrectNonce`-stamped scripts and modulepreloads got a nonce the active policy blocks. Same bug class #1049 fixed for the boundary/frame paths (`reconcileHeadMetas` excludes `META_KEY_CSP_NONCE`); `mergeHead` predates it. Fix: skip `csp-nonce` in both mergeHead loops.

## Closes #1051 - dev-warn when a streamed Suspense resolve is dropped

`applyStreamedResolve` silently drops a resolve whose boundary placeholder is absent. The drop is the correct designed posture (a superseded / degraded / discarded nav), and this PR does NOT change it (no retry, which would run outside the nav-token fence). But the silence is exactly what made the #1048 view-transition race take a full live-repro investigation. Add a dev-only, fire-once-per-id `warnDropped(id)` naming the boundary, mirroring the existing `devWarnFallback` dev-diagnostic. Never warns in production, never throws.

## Tests
- Unit (`packages/core/test/routing/router-client.test.js`): mergeHead keeps the original nonce (identity + content + a created script stamped with the original), no duplicate; the dropped-resolve warns once in dev, is silent on a successful resolve and in production. 185/185 pass.

## Definition-of-done surfaces
- Docs: N/A because neither change touches a public/authoring surface (a background-path correctness fix + a dev-only console diagnostic).
- Dogfood: website / docs / ui-website boot 200 with no broken preloads; blog e2e 78/78 (dist mode, rebuilt).
- Browser layer: N/A because both changes are DOM-head-merge + console logic fully modeled by the node/linkedom harness (no headline browser behavior beyond the streaming-resolve path #1049's browser suite already covers).
- Self-review loop converged (last round clean).

### CODE REVIEW

File: packages/core/src/router-client.js [hunk 1]
```diff
@@ -3937,13 +3955,23 @@ function mergeHead(newHead) {
     // headline #936 symptom. Keeping it is safe: a genuinely stale sheet is
     // dropped by the deploy-level hard reload (build-id mismatch), not here.
     if (isPersistentHeadStyle(el)) continue;
+    // Never remove the CSP nonce meta: the incoming full-body response carries a
+    // FRESH per-request nonce, but the browser enforces CSP against the nonce the
+    // ORIGINAL page load declared (see `getCspNonce`), so the live one must stay
+    // (#1050). `outerHTMLForDiff` strips the nonce ATTRIBUTE but not the `content`
+    // it lives in on this meta, so without this it looks "changed" and is dropped.
+    if (el.tagName === 'META' && metaIdentity(el) === META_KEY_CSP_NONCE) continue;
```
Comment (vivek7405) [inline packages/core/src/router-client.js:3963]: metaIdentity runs on every head child here, unlike reconcileHeadMetas which only iterates <meta>. A non-meta element with name=csp-nonce would be falsely skipped. Guard on tagName === 'META'.

File: packages/core/src/router-client.js [hunk 2]
```diff
@@ -780,6 +780,24 @@ function devWarnFallback(cause, href) {
...
+ * is what made the #1048 view-transition race hard to diagnose). Never warns in
+ * production, never throws.
+ *
+ * @param {string} id the streamed boundary id that could not be applied
+ */
+function warnDropped(id) {
+  if (typeof process !== 'undefined' && process.env && process.env.NODE_ENV === 'production') return;
+  warnOnce(
+    `stream-drop:${id}`,
+    `[webjs] dropped a streamed Suspense resolve for "${id}": no #${id} boundary in the DOM. Benign if this navigation was superseded or degraded. A stuck skeleton here means the shell swap did not place the boundary.`
```
Comment (vivek7405) [inline packages/core/src/router-client.js:797]: Semicolon used as a pause here (invariant 11); devWarnFallback's message avoids it. Use a period.

**Review by vivek7405 — COMMENTED** [review]
Read through both fixes. The mergeHead csp-nonce carve-out and the dev-warn are correct and the tests hold. Three small things worth tightening: the nonce skip is called on every head child without a tag guard, the warn message has a semicolon pause, and the mergeHead test only covered the both-present case. All three addressed.

File: packages/core/test/routing/router-client.test.js [hunk 3]
```diff
@@ -959,6 +1004,41 @@ test('mergeHead: applies meta csp-nonce to created scripts (replaces source nonc
...
+  const nonce = document.head.querySelectorAll('meta[name="csp-nonce"]');
+  assert.equal(nonce.length, 1, 'exactly one csp-nonce meta (not duplicated)');
+  assert.strictEqual(nonce[0], before, 'the original meta element is kept, not replaced');
+  assert.equal(nonce[0].getAttribute('content'), 'ORIGINAL', 'the original nonce content is preserved');
+  // A script created by the merge is stamped with the ORIGINAL nonce, not the incoming one.
+  assert.equal(document.head.querySelector('script[src="/after.js"]').getAttribute('nonce'), 'ORIGINAL',
+    'created scripts get the original page-load nonce');
+});
+
+test('mergeHead: keeps the csp-nonce meta when the incoming head omits it (#1050)', () => {
```
Comment (vivek7405) [inline packages/core/test/routing/router-client.test.js:1028]: The mergeHead csp-nonce coverage was only the both-present case; the incoming-omits-it path (removal loop must keep the original) was not asserted.

File: packages/core/src/router-client.js (hunk 1 above)
Comment (vivek7405) [reply packages/core/src/router-client.js:3963]: Fixed in [`3878df79`](https://github.com/webjsdev/webjs/commit/3878df79). Guarded both mergeHead loops on tagName === 'META'.

File: packages/core/src/router-client.js (hunk 2 above)
Comment (vivek7405) [reply packages/core/src/router-client.js:797]: Fixed in [`3878df79`](https://github.com/webjsdev/webjs/commit/3878df79). Rephrased to a period.

File: packages/core/test/routing/router-client.test.js (hunk 3 above)
Comment (vivek7405) [reply packages/core/test/routing/router-client.test.js:1028]: Fixed in [`3878df79`](https://github.com/webjsdev/webjs/commit/3878df79). Added the incoming-omits-csp-nonce case (original kept, same element).

**Review by vivek7405 — COMMENTED** [review]
Went back over the mergeHead carve-out with the tag guard, the warn gating, and the added test. All correct, no interaction with the #936 stylesheet preservation, counterfactuals hold. Nothing left.

**Review by vivek7405 — COMMENTED** [review]
One more pass over the whole diff. Confirmed the production gate actually holds in browsers (the server injects process.env.NODE_ENV browser-side, ssr.js:1197, and the gate is byte-identical to devWarnFallback), the carve-out composes with the addNewHeadElements/reconcileHeadMetas exclusions so no path adds or removes a csp-nonce meta on any swap tier, and the streamer reaches the warn only past the isCurrent fence, so it stays quiet on routine superseded navs. One semantics note, accepted deliberately: warnOnce keys per id per page session, and boundary ids recycle across pages, so a second distinct s1 drop in one session is silent. That matches devWarnFallback convention, and dev iteration reloads reset it, so the first occurrence after any reload always surfaces. Nothing to change.
